drivers/shared/executor/client.go

```diff
@@ -181,3 +183,62 @@ func (c *grpcExecutorClient) Exec(deadline time.Time, cmd string, args []string)
 
 	return resp.Output, int(resp.ExitCode), nil
 }
+
+func (d *grpcExecutorClient) ExecStreaming(ctx context.Context,
+	command []string,
+	tty bool,
+	execStream drivers.ExecTaskStream) error {
+
+	stream, err := d.client.ExecStreaming(ctx)
+	if err != nil {
+		return grpcutils.HandleGrpcErr(err, d.doneCtx)
+	}
+
+	err = stream.Send(&dproto.ExecTaskStreamingRequest{
+		Setup: &dproto.ExecTaskStreamingRequest_Setup{
+			Command: command,
+			Tty:     tty,
+		},
+	})
+	if err != nil {
+		return grpcutils.HandleGrpcErr(err, d.doneCtx)
+	}
+
+	errCh := make(chan error, 1)
+	go func() {
+		for {
+			m, err := execStream.Recv()
+			if err == io.EOF {
+				return
+			} else if err != nil {
+				errCh <- err
+				return
+			}
+
+			if err := stream.Send(m); err != nil {
+				errCh <- err
+				return
+			}
+
+		}
+	}()
+
+	for {
+		select {
+		case err := <-errCh:
+			return err
+		default:
+		}
+
+		m, err := stream.Recv()
+		if err == io.EOF {
+			return nil
+		} else if err != nil {
+			return err
+		}
+
+		if err := execStream.Send(m); err != nil {
+			return err
```

Comment:
Do all returns need grpcutils.HandleGrpcErr? I'm a little fuzzy on where that needs to be called.

If all returns do need that check maybe just wrap the whole method in a shim to do the check?

```go
func (d *grpcExecutorClient) ExecStreaming(...) error {
  if err := d.execStreaming(...); err != nil {
    return grpcutils.HandleGrpcErr(err, d.doneCtx)
  }
}

// actual impl  in execStreaming method
```

Reply:
we don't call `grpcutils.HandleGrpcErr` anywhere here.  It seems mostly for detecting shutdown and being able to handle it (by restarting, etc), in this case, we don't want to handle executor shutdown any specially.

Reply:
Ah! execStream is "this" (the driver) side whereas the HandleGrpcErr calls above are for remote calls to executors? I guess I'm still confused why Recv wouldn't call HandleGrpcErr as its the executor side...

Reply:
Oh I see - I missed it - i'll add it.